SimCalorimetry/EcalEBTrigPrimProducers/plugins/EcalEBTrigPrimPhase2Producer.cc

```diff
@@ -0,0 +1,185 @@
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+#include "DataFormats/Provenance/interface/ProductID.h"
+#include "DataFormats/Provenance/interface/ParameterSetID.h"
+#include "DataFormats/Provenance/interface/Provenance.h"
+#include "DataFormats/EcalDigi/interface/EcalDigiCollections.h"
+#include "CondFormats/EcalObjects/interface/EcalLiteDTUPedestals.h"
+#include "CondFormats/DataRecord/interface/EcalLiteDTUPedestalsRcd.h"
+
+//#include "CondFormats/EcalObjects/interface/EcalTPGTowerStatus.h"
+//#include "CondFormats/DataRecord/interface/EcalTPGStripStatusRcd.h"
+//#include "CondFormats/EcalObjects/interface/EcalTPGStripStatus.h"
+
+#include "EcalEBTrigPrimPhase2Producer.h"
+
+#include <memory>
+
+#include "SimCalorimetry/EcalEBTrigPrimAlgos/interface/EcalEBPhase2TrigPrimAlgo.h"
+
+EcalEBTrigPrimPhase2Producer::EcalEBTrigPrimPhase2Producer(const edm::ParameterSet& iConfig)
+    : barrelOnly_(iConfig.getParameter<bool>("BarrelOnly")),
+      debug_(iConfig.getParameter<bool>("Debug")),
+      famos_(iConfig.getParameter<bool>("Famos")),
+      binOfMaximum_(iConfig.getParameter<int>("binOfMaximum")) {
+  tokenEBdigi_ = consumes<EBDigiCollectionPh2>(iConfig.getParameter<edm::InputTag>("barrelEcalDigis"));
+
+  eTTmapToken_ = esConsumes<edm::Transition::BeginRun>();
+  theGeometryToken_ = esConsumes<edm::Transition::BeginRun>();
+
+  theEcalTPGPedestals_Token_ =
+      esConsumes<EcalLiteDTUPedestalsMap, EcalLiteDTUPedestalsRcd, edm::Transition::BeginRun>();
+  theEcalEBPhase2TPGPedestals_Token_ =
+      esConsumes<EcalEBPhase2TPGPedestalsMap, EcalEBPhase2TPGPedestalsRcd, edm::Transition::BeginRun>();
+
+  theEcalTPGCrystalStatus_Token_ =
+      esConsumes<EcalTPGCrystalStatus, EcalTPGCrystalStatusRcd, edm::Transition::BeginRun>();
+  theEcalEBPhase2TPGLinearization_Token_ =
+      esConsumes<EcalEBPhase2TPGLinearizationConst, EcalEBPhase2TPGLinearizationConstRcd, edm::Transition::BeginRun>();
+  theEcalEBTPGAmplWeightIdMap_Token_ =
+      esConsumes<EcalEBPhase2TPGAmplWeightIdMap, EcalEBPhase2TPGAmplWeightIdMapRcd, edm::Transition::BeginRun>();
+  theEcalEBTPGTimeWeightIdMap_Token_ =
+      esConsumes<EcalEBPhase2TPGTimeWeightIdMap, EcalEBPhase2TPGTimeWeightIdMapRcd, edm::Transition::BeginRun>();
+  theEcalTPGWeightGroup_Token_ = esConsumes<EcalTPGWeightGroup, EcalTPGWeightGroupRcd, edm::Transition::BeginRun>();
+
+  //register your products
+  produces<EcalEBPhase2TrigPrimDigiCollection>();
+}
+
+void EcalEBTrigPrimPhase2Producer::beginRun(edm::Run const& run, edm::EventSetup const& setup) {
+  auto const& theGeometry = setup.getData(theGeometryToken_);
+  auto const& eTTmap = setup.getData(eTTmapToken_);
+
+  algo_ = std::make_unique<EcalEBPhase2TrigPrimAlgo>(&eTTmap, &theGeometry, binOfMaximum_, debug_);
+
+  // get a first version of the records
+  cacheID_ = this->getRecords(setup);
+
+  nEvent_ = 0;
```

Comment:
This could be set in the initializer list of the constructor.

Reply:
Prefer to keep  it in the beginRun